### PR TITLE
Fix MySQL WriteRelationships query

### PR DESCRIPTION
**Pull request by trrrrrys**
In the WriteRelationships TOUCH process for MySQL, if a tuple to be written already exists and is marked as deleted, it doesn't execute an INSERT. Additionally, if multiple deleted tuples exist, the deleted_transaction is updated by the TOUCH operation.

I have made modifications by adding a WHERE clause to enable UPSERT operations.

### CODE REVIEW

**Comment by trrrrrys** [comment]
I have read the CLA Document and I hereby sign the CLA

**Comment by trrrrrys** [comment]
I have read the CLA Document and I hereby sign the CLA

**Comment by trrrrrys** [comment]
recheck

**Comment by josephschorr** [comment]
@trrrrrys Please add tests for the changes; I'm not 100% certain what the underlying bug(s) are

**Comment by trrrrrys** [comment]
@josephschorr 
I added a test.
When using WriteRelationships to CREATE a relation and then DELETE and TOUCH it, Upsert does not get executed.

File: pkg/datastore/test/tuples.go [hunk 1]
```diff
@@ -598,6 +598,35 @@ func TouchAlreadyExistingTest(t *testing.T, tester DatastoreTester) {
...
+	ds, _ := testfixtures.StandardDatastoreWithData(rawDS, require)
+	ctx := context.Background()
+
+	tpl1 := makeTestTuple("foo", "tom")
+	tpl2 := makeTestTuple("foo", "sarah")
+
+	_, err = common.WriteTuples(ctx, ds, core.RelationTupleUpdate_CREATE, tpl1, tpl2)
+	require.NoError(err)
+
+	ensureTuples(ctx, require, ds, tpl1, tpl2)
```
Comment (josephschorr) [inline pkg/datastore/test/tuples.go:617]: Please add a TOUCH here and ensure the tuples are still present

Comment (trrrrrys) [reply pkg/datastore/test/tuples.go:617]: @josephschorr 
Does that mean the tests are executed in the order of CREATE, TOUCH, DELETE, TOUCH?

```go
// WriteDeleteTouchTest tests writing a relationship, deleting it, and then touching it.
func WriteDeleteTouchTest(t *testing.T, tester DatastoreTester) {
	require := require.New(t)

	rawDS, err := tester.New(0, veryLargeGCInterval, veryLargeGCWindow, 1)
	require.NoError(err)

	ds, _ := testfixtures.StandardDatastoreWithData(rawDS, require)
	ctx := context.Background()

	tpl1 := makeTestTuple("foo", "tom")
	tpl2 := makeTestTuple("foo", "sarah")

	_, err = common.WriteTuples(ctx, ds, core.RelationTupleUpdate_CREATE, tpl1, tpl2)
	require.NoError(err)

	ensureTuples(ctx, require, ds, tpl1, tpl2)

        // Add TOUCH Test
	_, err = common.WriteTuples(ctx, ds, core.RelationTupleUpdate_TOUCH, tpl1, tpl2)
	require.NoError(err)

	ensureTuples(ctx, require, ds, tpl1, tpl2)
        // 

	_, err = common.WriteTuples(ctx, ds, core.RelationTupleUpdate_DELETE, tpl1, tpl2)
	require.NoError(err)

	ensureNotTuples(ctx, require, ds, tpl1, tpl2)

	_, err = common.WriteTuples(ctx, ds, core.RelationTupleUpdate_TOUCH, tpl1, tpl2)
	require.NoError(err)

	ensureTuples(ctx, require, ds, tpl1, tpl2)
}
```

Comment (josephschorr) [reply pkg/datastore/test/tuples.go:617]: Yeah, I'm saying add another test with CREATE, TOUCH, DELETE, TOUCH in addition to one with CREATE, DELETE, TOUCH

Comment (trrrrrys) [reply pkg/datastore/test/tuples.go:617]: @josephschorr 
Added test. Could you check it?
[8dae584](https://github.com/authzed/spicedb/pull/1478/commits/8dae584db2e3fa4ad5f4a715c412b3330f060bc4)

Comment (josephschorr) [reply pkg/datastore/test/tuples.go:617]: Running

**Review by ecordell — APPROVED** [review]

**Pull request merged**